Bio/SearchIO/InfernalIO/infernal_tab.py

```diff
@@ -0,0 +1,363 @@
+
+from Bio.SearchIO._index import SearchIndexer
+from Bio.SearchIO._model import HSP
+from Bio.SearchIO._model import HSPFragment
+from Bio.SearchIO._model import QueryResult
+
+from ._base import _BaseInfernalParser
+
+__all__ = ("InfernalTabParser", "InfernalTabIndexer")
+
+# tabular format column names
+_TAB_FORMAT = {
+    1: (
+        "target_name",
+        "target_acc",
+        "query_name",
+        "query_acc",
+        "mdl",
+        "mdl_from",
+        "mdl_to",
+        "seq_from",
+        "seq_to",
+        "strand",
+        "trunc",
+        "pass",
+        "gc",
+        "bias",
+        "score",
+        "evalue",
+        "inc",
+        "description",
+    ),
+    2: (
+        "idx",
+        "target_name",
+        "target_acc",
+        "query_name",
+        "query_acc",
+        "clan",
+        "mdl",
+        "mdl_from",
+        "mdl_to",
+        "seq_from",
+        "seq_to",
+        "strand",
+        "trunc",
+        "pass",
+        "gc",
+        "bias",
+        "score",
+        "evalue",
+        "inc",
+        "olp",
+        "anyidx",
+        "afrct1",
+        "afrct2",
+        "winidx",
+        "wfrct1",
+        "wfrct2",
+        "mdl_len",
+        "seq_len",
+        "description",
+    ),
+    3: (
+        "target_name",
+        "target_acc",
+        "query_name",
+        "query_acc",
+        "mdl",
+        "mdl_from",
+        "mdl_to",
+        "seq_from",
+        "seq_to",
+        "strand",
+        "trunc",
+        "pass",
+        "gc",
+        "bias",
+        "score",
+        "evalue",
+        "inc",
+        "mdl_len",
+        "seq_len",
+        "description",
+    ),
+}
+_DEFAULT_TAB_FORMAT = 1
+
+# column to class attribute map
+_COLUMN_QRESULT = {
+    "query_name": ("id", str),
+    "query_acc": ("accession", str),
+    "seq_len": ("seq_len", int),
+    "clan": ("clan", str),
+}
+_COLUMN_HIT = {
+    "target_name": ("id", str),
+    "target_acc": ("accession", str),
+    "query_name": ("query_id", str),
+    "description": ("description", str),
+    "mdl_len": ("seq_len", int),
+}
+_COLUMN_HSP = {
+    "score": ("bitscore", float),
+    "evalue": ("evalue", float),
+    "bias": ("bias", float),
+    "gc": ("gc", float),
+    "mdl": ("model", str),
+    "trunc": ("truncated", str),
+    "pass": ("pipeline_pass", int),
+    "inc": ("is_included", str),
+    "olp": ("olp", str),
+    "anyidx": ("anyidx", str),
+    "afrct1": ("afrct1", str),
+    "afrct2": ("afrct2", str),
+    "winidx": ("winidx", str),
+    "wfrct1": ("wfrct1", str),
+    "wfrct2": ("wfrct2", str),
+}
+_COLUMN_FRAG = {
+    "mdl_from": ("query_start", int),
+    "mdl_to": ("query_end", int),
+    "seq_from": ("hit_start", int),
+    "seq_to": ("hit_end", int),
+    "strand": ("hit_strand", str),
+}
+
+
+class InfernalTabParser(_BaseInfernalParser):
+    """Parser for the Infernal tabular format."""
+
+    def __init__(self, handle, fmt=_DEFAULT_TAB_FORMAT):
```

Comment:
From a UI point of view, and without being familiar with Infernal, having a main SearchIO ``format`` argument (here ``infernal-tab`` I believe) and a secondary ``fmt`` argument seems confusing.

Does it make sense to think about ``format="infernal-tab1"``, ``format="infernal-tab2"`` etc?